Libraries/rocSOLVER/syev_batched/main.cpp

```diff
@@ -0,0 +1,264 @@
+
+    const rocblas_int full_size = batch_size * single_matrix_num_elements * sizeof(rocblas_double*);
+    const rocblas_int batch_eigenvalue_size = sizeof(double) * batch_eigenvalue_num_elements;
+
+    HIP_CHECK(hipMalloc(&d_info, info_size));
+    HIP_CHECK(hipMalloc(&d_A, full_size));
+    HIP_CHECK(hipMalloc(&d_W, batch_eigenvalue_size));
+    HIP_CHECK(hipMalloc(&array_dA, array_size));
+
+    HIP_CHECK(hipMemcpy(d_A, A.data(), full_size, hipMemcpyHostToDevice));
+    for(int k = 0; k < batch_size; ++k)
+    {
+        array_A[k] = &d_A[k * single_matrix_num_elements];
+    }
+    HIP_CHECK(hipMemcpy(array_dA, array_A, array_size, hipMemcpyHostToDevice));
+
+    // 6. Initialize rocBLAS.
+    rocblas_handle handle;
+    ROCBLAS_CHECK(rocblas_create_handle(&handle));
+
+    // 7. Get and reserve the working space on device.
+    const rocblas_int batch_tridiagonal_size
+        = batch_eigenvalue_num_elements * sizeof(rocblas_double);
+    rocblas_double* d_work = nullptr;
+    HIP_CHECK(hipMalloc(&d_work, batch_tridiagonal_size));
+
+    // 8. Compute eigenvectors and eigenvalues.
+    ROCBLAS_CHECK(rocsolver_dsyev_batched(handle,
+                                          evect,
+                                          uplo,
+                                          n,
+                                          array_dA,
+                                          lda,
+                                          d_W,
+                                          batch_size,
+                                          d_work,
+                                          n * n,
+                                          d_info,
+                                          batch_size));
+
+    // 9. Get results from device.
+    HIP_CHECK(hipMemcpy(V.data(), d_A, full_size, hipMemcpyDeviceToHost));
+    HIP_CHECK(hipMemcpy(W.data(), d_W, batch_eigenvalue_size, hipMemcpyDeviceToHost));
+    HIP_CHECK(hipMemcpy(info.data(), d_info, info_size, hipMemcpyDeviceToHost));
+
+    // 10. Clean up device allocations.
+    ROCBLAS_CHECK(rocblas_destroy_handle(handle));
+    HIP_CHECK(hipFree(d_A));
+    HIP_CHECK(hipFree(d_W));
+    HIP_CHECK(hipFree(d_work));
+    HIP_CHECK(hipFree(d_info));
+
+    // 11. Print results.
+    auto print_results = [&](rocblas_int batch_id)
+    {
+        if(info[batch_id] == 0)
+        {
+            std::cout << "SYEV converges." << std::endl;
+        }
+        else if(info[batch_id] > 0)
+        {
+            std::cout << "SYEV does not converge (" << info[batch_id]
+                      << " elements did not converge)." << std::endl;
+        }
+
+        std::cout << "\nGiven the n x n square input matrix A; we computed the orthonormal "
+                     "eigenvectors V and the associated eigenvalues W."
+                  << std::endl;
+        auto A_begin = A.begin();
+        auto W_begin = W.begin();
+        auto V_begin = V.begin();
```

Comment:
This is only valid for the first batch instance (i.e. batch_id = 0)

Reply:
Indeed. Fixed it!